plug.vim

```diff
@@ -116,6 +116,104 @@ let s:TYPE = {
 let s:loaded = get(s:, 'loaded', {})
 let s:triggers = get(s:, 'triggers', {})
 
+function! s:isabsolute(dir) abort
+  return a:dir =~# '^/' || (has('win32') && a:dir =~? '^\%(\\\|[A-Z]:\)')
+endfunction
+
+function! s:get_gitdir(dir) abort
+  let l:gitdir = a:dir . '/.git'
+  if isdirectory(l:gitdir)
+    return l:gitdir
+  endif
+  try
+    let l:line = readfile(l:gitdir)[0]
+    if l:line =~# '^gitdir: '
+      let l:gitdir = l:line[8:]
+      if !s:isabsolute(l:gitdir)
+        let l:gitdir = a:dir . '/' . l:gitdir
+      endif
+      if isdirectory(l:gitdir)
+        return l:gitdir
+      endif
+    endif
+  catch
+  endtry
+  return ''
+endfunction
+
+function! s:git_get_remote_origin_url(dir) abort
+  let gitdir = s:get_gitdir(a:dir)
+  if gitdir ==# ''
+    return v:null
```

Comment:
Using `v:null` requires Vim 8.0. Is it acceptable for vim-plug?
Minpac doesn't have this issue, because its requirement is Vim 8.0 or later.
(Same for `s:git_get_revision()` and `s:git_get_branch()`.)